pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/SystemTopic.java

```diff
@@ -70,8 +72,11 @@ public void checkGC() {
     }
 
     @Override
-    public CompletableFuture<Boolean> isCompactionEnabled() {
-        // All system topics are using compaction, even though is not explicitly set in the policies.
-        return CompletableFuture.completedFuture(true);
+    public boolean isCompactionEnabled() {
+        // All system topics are using compaction except `HealthCheck`,
+        // even though is not explicitly set in the policies.
+        TopicName name = TopicName.get(topic);
+        NamespaceName heartbeatNamespace = brokerService.pulsar().getHeartbeatNamespaceV2();
```

Comment:
Should we be compatible with versions prior to v2?

Reply:
Yes, right. Updated.